xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/DXVAHD.h

```diff
@@ -104,6 +105,7 @@ class CProcessorHD : public ID3DResource
   D3D11_VIDEO_PROCESSOR_RATE_CONVERSION_CAPS m_rateCaps;
   D3D11_TEXTURE2D_DESC m_texDesc;
   PROCESSOR_VIEW_TYPE m_eViewType;
+  std::vector<AVPixelFormat> m_formats;
```

Comment:
simply call the static method again after adapters have changed. I hooked vaapi/vdpau into RefreshContext: https://github.com/FernetMenta/kodi-agile/blob/videobuffer/xbmc/windowing/X11/WinSystemX11GLContext.cpp#L213